src/main/java/com/code_intelligence/jazzer/mutation/mutator/lang/FloatingPointMutatorFactory.java

```diff
@@ -40,31 +52,203 @@ public Optional<SerializingMutator<?>> tryCreate(AnnotatedType type, MutatorFact
+      }
+      float[] specialValuesArray = new float[specialValuesList.size()];
+      for (int i = 0; i < specialValuesList.size(); i++) {
+        specialValuesArray[i] = specialValuesList.get(i);
+      }
+      return specialValuesArray;
+    }
 
     public Float mutateWithLibFuzzer(Float value) {
       return LibFuzzerMutator.mutateDefault(value, this, 0);
     }
 
-    // TODO: randomly init to special values like NaN, -0, +0, +Inf, -Inf
     @Override
     public Float init(PseudoRandom prng) {
-      return prng.closedRange(-Float.MAX_VALUE, Float.MAX_VALUE);
+      if (prng.choice()) {
+        return specialValues[prng.closedRange(0, specialValues.length - 1)];
+      } else {
+        return prng.closedRange(minValue, maxValue);
+      }
     }
 
-    // TODO: consider current value when mutating
     @Override
     public Float mutate(Float value, PseudoRandom prng) {
-      return prng.closedRange(-Float.MAX_VALUE, Float.MAX_VALUE);
+      float result;
+      // small chance to return a special value
+      if (prng.closedRange(0, INV_CHANCE_SPECIAL_VALUE) == 0) {
+        result = specialValues[prng.closedRange(0, specialValues.length - 1)];
+      } else {
+        switch (prng.closedRange(0, 5)) {
+          case 0:
+            result = mutateWithBitFlip(value, prng);
+            break;
+          case 1:
+            result = mutateExponent(value, prng);
+            break;
+          case 2:
+            result = mutateMantissa(value, prng);
+            break;
+          case 3:
+            result = (float) mutateWithMathematicalFn(value, prng);
+            break;
+          case 4:
+            result = mutateWithLibFuzzer(value);
+            break;
+          default: // random in range cannot exceed the given bounds (and cannot be NaN)
+            return (prng.closedRange(minValue, maxValue));
+        }
+      }
+      result = forceInRange(result, minValue, maxValue, allowNaN);
+      // If value repeats, or is a not allowed NaN, mutate again
+      if (result == value || (Float.isNaN(result) && Float.isNaN(value))
+          || (Float.isNaN(result) && !allowNaN)) {
+        return mutate(value, prng);
+      }
+      return result;
+    }
+
+    static float forceInRange(float value, float minValue, float maxValue, boolean allowNaN) {
+      if ((value >= minValue && value <= maxValue) || (Float.isNaN(value) && allowNaN)) {
+        return value;
+      }
+
+      // Clamp infinite values
+      if (value == Float.POSITIVE_INFINITY)
+        return maxValue;
+      if (value == Float.NEGATIVE_INFINITY)
+        return minValue;
+
+      // If NaN was allowed, it was handled above.
+      // Here we replace NaN by the middle of the clamped finite range.
+      if (Float.isNaN(value)) {
+        // maxValue or minValue may be infinite, so we need to clamp them.
+        return minValue
+            + (Math.min(Float.MAX_VALUE, maxValue) * 0.5f
+                - Math.max(Float.MIN_VALUE, minValue) * 0.5f);
```

Comment:
I think that just `Math.min(Float.MAX_VALUE, maxValue) * 0.5f + Math.max(Float.MIN_VALUE, minValue) * 0.5f` should be equally safe in terms of overflows.